app/controllers/api/order_details_controller.rb

```diff
@@ -4,22 +4,37 @@ class Api::OrderDetailsController < ApplicationController
   http_basic_authenticate_with name: Settings.api.basic_auth_name, password: Settings.api.basic_auth_password
 
   def show
-    od = OrderDetail.find(params[:id])
-    account = od.account
-    owner = account.owner.user
-    ordered_for = od.order.user
-
-    render json: {
-      account: {
-        id: account.id,
-        owner: user_to_hash(owner),
-      },
-      ordered_for: user_to_hash(ordered_for),
-    }
+    render json: response_to_hash
   end
 
   private
 
+  def account
+    @account = order_detail.try(:account) unless defined?(@account)
+    @account
+  end
+
+  def account_to_hash
+    if account.present?
+      { id: account.id, owner: user_to_hash(account.owner.user) }
+    end
+  end
+
+  def ordered_for_to_hash
+    if order_detail.present?
+      user_to_hash(order_detail.order.user)
+    end
+  end
+
+  def response_to_hash
+    { account: account_to_hash, ordered_for: ordered_for_to_hash }
+  end
+
+  def order_detail
+    @order_detail = OrderDetail.find_by_id(params[:id]) unless defined?(@order_detail)
```

Comment:
You're right. It's really no different from `@order_detail ||= OrderDetail.find_by_id(params[:id])` then.

As for a 404 vs. a 200 with an empty return, either will have the same result on the client end: no attributes updated (or it will, after https://github.com/tablexi/nucore-imserc/issues/60 goes live).